### PR TITLE
[ROCm] GPU unit test fix for gfx90a specific failure.

### PR DESCRIPTION
**Pull request by rsanthanam-amd**
/cc @cheshire

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc [hunk 1]
```diff
@@ -70,16 +70,26 @@ TEST_F(GpuUnrollingTest, UnrollFourTimes) {
   config.set_debug_options(debug_options);
   auto hlo_module = ParseAndReturnVerifiedModule(kAddModule, config).value();
 
-  CompileAndVerifyIr(std::move(hlo_module),
-                     R"(
+  const std::string pattern1 = R"(
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc:73]: Is the issue that the vectorization occurs only with this hardware? Shouldn't we be matching unoptimized LLVM in this case?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc (hunk 1 above)
Comment (rsanthanam-amd) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc]: Fixed it for the UnrollFourTimes subtest but that did not work for the other subtests.

Comment (rsanthanam-amd) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc]: For those other tests, the following IR line is failing:

; CHECK: load <4 x float>

**Comment by rsanthanam-amd** [comment]
@cheshire gentle ping

File: tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc (hunk 1 above)
Comment (rsanthanam-amd) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc]: I changed the test a bit to make the matching of unoptimized LLVM work.  PTAL.

Comment (rsanthanam-amd) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc]: I also discovered that the DisabledUnrollUnfusedSine and DisabledUnrollUnfusedCosine no longer call ROCm-Device-Lib functions but now generate LLVM so I updated those tests to remove the ROCm specific IR.

**Review by cheshire — APPROVED** [review]

File: tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc (hunk 1 above)
Comment (rsanthanam-amd) [reply tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc]: Sorry, I pushed one more change to fix a typo and alter one of the subtests.

**Pull request closed without merging**